### PR TITLE
Add pull request template

### LINKED ISSUE #1394: [MAINTENANCE] changelog helper bot
## Description

Develop a simple GitHub webhooks bot that receives PR commits, and if it can't find a changelog modification in the PR make a comment on the PR. It would look something like this:

> I noticed this PR doesn't have a changelog entry, if it should have one you can use this
```
* BOT INSERTS PR SUBJECT HERE ([#BOT INSERTS PR NUMBER](BOT INSERTS PR URL))
```
> I'm a robot, if this isn't helpful please let the amethyst organization know.

## Reason

Forgetting changelog entries is oftentimes an innocent mistake. This can help mitigate that, and give us greater confidence in our changelog.

## Impact

For smaller PRs this will be a bit of a nuisance. We can mitigate this by limiting the bot to one comment per PR.

## Additional Information

I can host this bot on my free google cloud micro instance.

### PR DESCRIPTION
**Pull request by Xaeroxe**
This should help us with keeping future development more disciplined.

Fixes #1394

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+- List API removals here
+
+## Modifications
+
+- List changes to existing structures and functions here
+
+## PR Checklist
+
+By placing an x in the boxes I certify that I have
```
Comment (khionu) [inline .github/pull_request_template.md]: With this phrasing, it suggests the following list are part of the full sentence. Either need to change this line so the checkboxes are not grammatically dependent on this, or change the checkboxes so they match the grammar flow. The former would be easier, but if possible, the latter would be more professional.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+## Description
+
+Add a brief summary of your PR here, between 1 sentence and two paragraphs
```
Comment (khionu) [inline .github/pull_request_template.md]: Need to spell out "one", here. The really formal rule is to spell out numbers 10 or less. Alternatively, use the numeric "2". I would go with the former.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Modifications
+
+- List changes to existing structures and functions here
+
+## PR Checklist
+
+By placing an x in the boxes I certify that I have
+
+- [ ] Run `cargo test --all` locally
```
Comment (khionu) [inline .github/pull_request_template.md]: Forgot to mention, this should be "ran", sorry

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+- List API removals here if any
+
+## Modifications
+
+- List changes to existing structures and functions here if any
+
+## PR Checklist
+
+By placing an x in the boxes I certify that I have
```
Comment (azriel91) [inline .github/pull_request_template.md]: For punctuation consistency, can you add a `:` here, and a fullstop at the end of every bullet point?

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+- Detail API additions here if any
+- Do not list changes or removals
+
+## Removals
+
+- List API removals here if any
+
+## Modifications
+
+- List changes to existing structures and functions here if any
```
Comment (azriel91) [inline .github/pull_request_template.md]: I'm thinking we shouldn't need to list these here, since they should go in the changelog

Comment (Xaeroxe) [reply .github/pull_request_template.md]: Having it in both places can be advantageous for reviewers. Ideally it'd just be some copy pastes anyway I think.

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+## PR Checklist
+
+By placing an x in the boxes I certify that I have:
+
+- [ ] Ran `cargo test --all` locally if this modified any rs files.
+- [ ] Ran `cargo fmt --all` locally if this modified any rs files.
```
Comment (jojolepro) [inline .github/pull_request_template.md]: cargo +stable fmt --all

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] Updated the content of the book if this PR would make the book outdated.
+- [ ] Added a changelog entry if this will impact users.
```
Comment (jojolepro) [inline .github/pull_request_template.md]: Even if it doesn't impact, we prefer when there is one, if the change is > 5 lines and is not documentation.

**Comment by Xaeroxe** [comment]
By the way since this is a large change that impacts the entire org I am not planning on merging this PR quickly. I want to make sure we go through the waking hours of all org members first.

**Comment by torkleyy** [comment]
Can we add “Added tests for new code“ to the checklist?

**Comment by Xaeroxe** [comment]
Testing a PR that adds a new md file would be a waste of CI resources, so I'm going to skip bors for this one.

**Comment by LucioFranco** [comment]
@Xaeroxe sounds like we need integration tests for our md files!!!!! 😄

**Pull request merged**